plugins/base/src/main/kotlin/transformers/documentables/InheritorsExtractorTransformer.kt

```diff
@@ -0,0 +1,76 @@
+                is Package -> copy(classlikes = classlikes.map { it.appendInheritors(inheritanceMap) as Classlike })
+                is Class -> copy(
+                    extra = extra + info,
+                    classlikes = classlikes.map { it.appendInheritors(inheritanceMap) as Classlike })
+                is Enum -> copy(
+                    extra = extra + info,
+                    classlikes = classlikes.map { it.appendInheritors(inheritanceMap) as Classlike })
+                is Interface -> copy(
+                    extra = extra + info,
+                    classlikes = classlikes.map { it.appendInheritors(inheritanceMap) as Classlike })
+                is Object -> copy(classlikes = classlikes.map { it.appendInheritors(inheritanceMap) as Classlike })
+                else -> this
+            }
+        }
+
+    private fun Map<PlatformData, Map<DRI, List<DRI>>>.getForDRI(dri: DRI) = map { (v, k) ->
+        v to k[dri]
+    }.map { (k, v) -> k to v.orEmpty() }.toMap().toPlatformDependent()
+
+    private fun <T> T.generateInheritanceMap(): Map<PlatformData, Map<DRI, List<DRI>>> where T : Documentable =
+        getInheritanceEntriesRec().filterNot { it.second.isEmpty() }.groupBy({ it.first }) { it.second }
+            .map { (k, v) ->
+                k to v.flatMap { p -> p.groupBy({ it.first }) { it.second }.toList() }
+                    .groupBy({ it.first }) { it.second }.map { (k2, v2) -> k2 to v2.flatten() }.toMap()
+            }.toMap()
+
+    private fun <T> T.getInheritanceEntriesRec(): List<Pair<PlatformData, List<Pair<DRI, DRI>>>> where T : Documentable =
+        this.toInheritanceEntries() + children.flatMap { it.getInheritanceEntriesRec() }
+
+    private fun <T> T.toInheritanceEntries(): List<Pair<PlatformData, List<Pair<DRI, DRI>>>> where T : Documentable =
+        this.let { it as? WithSupertypes }?.let {
+            it.supertypes.map.map { (k, v) -> k to v.map { it to dri }.toList() }.toList()
+        }.orEmpty()
+
+    fun PlatformDependent<List<DRI>>.toInheritorsInfo() = InheritorsInfo(this)
+}
+
+class InheritorsInfo(val value: PlatformDependent<List<DRI>>) : ExtraProperty<Documentable>, MapUtils {
+    companion object : ExtraProperty.Key<Documentable, InheritorsInfo> {
+        override fun mergeStrategyFor(left: InheritorsInfo, right: InheritorsInfo): MergeStrategy<Documentable> =
+            MergeStrategy.Replace(InheritorsInfo(PlatformDependent(left.value.map + right.value.map)))
+    }
+
+    override val key: ExtraProperty.Key<Documentable, *> = InheritorsInfo
+}
+
+interface MapUtils {
+
+    fun <T> Map<PlatformData, T>.toPlatformDependent(): PlatformDependent<T> = PlatformDependent(this)
+
+    operator fun <K, V> Map<K, List<V>>.plus(other: Map<K, List<V>>) =
+        (entries.toList() + other.entries.toList()).merge()
+
+    fun <K, V> List<Map.Entry<K, List<V>>>.merge(): Map<K, List<V>> =
+        groupBy({ it.key }) { it.value }.map { (k, v) -> k to v.flatten() }.toMap()
+
+}
```

Comment:
This interface seems unsecessary. Both `plus` and `merge` seems to have no usages and `toPlatformDependent` does nothing reasonable that couldn't be inlined